src/main/java/org/apache/sysml/hops/rewrite/ProgramRewriter.java

```diff
@@ -110,9 +110,11 @@ public ProgramRewriter( boolean staticRewrites, boolean dynamicRewrites )
 				_sbRuleSet.add(  new RewriteRemoveUnnecessaryBranches()          ); //dependency: constant folding		
 				_sbRuleSet.add(  new RewriteMergeBlockSequence()                 ); //dependency: remove branches
  			}
- 			if( OptimizerUtils.ALLOW_SPLIT_HOP_DAGS )
+ 			if( OptimizerUtils.ALLOW_SPLIT_HOP_DAGS ) {
  				_sbRuleSet.add(  new RewriteSplitDagUnknownCSVRead()             ); //dependency: reblock, merge blocks	
- 			if( ConfigurationManager.getCompilerConfigFlag(ConfigType.ALLOW_INDIVIDUAL_SB_SPECIFIC_OPS) )
+ 			        _sbRuleSet.add(new RewriteSplitDagFunctionOp()                   ); //dependency: ...
```

Comment:
let's handle this together with IPA because we only need to split the DAGs if we cannot propagate input to output sizes for these single-output functions.